pkg/orchestrator/data_scrubber.go

```diff
@@ -0,0 +1,127 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build orchestrator
+
+package orchestrator
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+)
+
+var (
+	defaultSensitiveWords = []string{
+		"password", "passwd", "mysql_pwd",
+		"access_token", "auth_token",
+		"api_key", "apikey", "pwd",
+		"secret", "credentials", "stripetoken"}
+)
+
+// DataScrubber allows the agent to blacklist cmdline arguments that match
+// a list of predefined and custom words
+type DataScrubber struct {
+	Enabled                  bool
+	RegexSensitivePatterns   []*regexp.Regexp
+	LiteralSensitivePatterns []string
+	scrubbedCmdlines         map[string][]string
+}
+
+// NewDefaultDataScrubber creates a DataScrubber with the default behavior: enabled
+// and matching the default sensitive words
+func NewDefaultDataScrubber() *DataScrubber {
+	newDataScrubber := &DataScrubber{
+		Enabled:                  true,
+		LiteralSensitivePatterns: defaultSensitiveWords,
+		scrubbedCmdlines:         make(map[string][]string),
+	}
+
+	return newDataScrubber
+}
+func (ds *DataScrubber) ContainsBlacklistedWord(word string) bool {
```

Comment:
did you see any case (in the wild or in unit tests) where this fails to find some words that we currently scrub (and thus would fail to scrub with the new function)

Reply:
I copied over all the test cases and they work just fine. 
If I remove the custom regex part the test cases which were using those will fail e.g. those:
https://github.com/DataDog/datadog-agent/blob/1bf82d4c25b1dc5a1b3fb4e69641e54819d6977a/pkg/orchestrator/data_scrubber_test.go#L122-L142